src/include/daos/object.h

```diff
@@ -341,37 +365,34 @@ daos_oclass_is_ec(struct daos_oclass_attr *oca)
 }
 
 static inline void
-daos_obj_set_oid(daos_obj_id_t *oid, daos_ofeat_t ofeats,
-		 daos_oclass_id_t cid, uint32_t args)
+daos_obj_set_oid(daos_obj_id_t *oid, enum daos_obj_type type,
+		 enum daos_obj_redun ord, uint32_t nr_grps,
+		 uint32_t args)
 {
 	uint64_t hdr;
 
-	/* TODO: add check at here, it should return error if user specified
-	 * bits reserved by DAOS
-	 */
+	/** XXX: encode nr_grps as-is for now */
+
 	oid->hi &= (1ULL << OID_FMT_INTR_BITS) - 1;
 	/**
 	 * | Upper bits contain
-	 * | OID_FMT_VER_BITS (version)		 |
-	 * | OID_FMT_FEAT_BITS (object features) |
+	 * | OID_FMT_TYPE_BITS (object features) |
 	 * | OID_FMT_CLASS_BITS (object class)	 |
-	 * | 96-bit for upper layer ...		 |
+	 * | OID_FMT_MD_BITS (object metadata)	 |
+	 * | 96-bit for API user ...		 |
 	 */
-	hdr  = ((uint64_t)OID_FMT_VER << OID_FMT_VER_SHIFT);
-	hdr |= ((uint64_t)ofeats << OID_FMT_FEAT_SHIFT);
-	hdr |= ((uint64_t)cid << OID_FMT_CLASS_SHIFT);
+	hdr  = ((uint64_t)type << OID_FMT_TYPE_SHIFT);
+	hdr |= ((uint64_t)ord << OID_FMT_CLASS_SHIFT);
+	hdr |= ((uint64_t)nr_grps << OID_FMT_MD_SHIFT);
 	oid->hi |= hdr;
 }
 
 /* check if an object ID is OIT (Object ID Table) */
 static inline bool
 daos_oid_is_oit(daos_obj_id_t oid)
 {
-	daos_oclass_id_t	oc = daos_obj_id2class(oid);
-
-	return (oc == DAOS_OC_OIT_RF0) || (oc == DAOS_OC_OIT_RF1) ||
-	       (oc == DAOS_OC_OIT_RF2) || (oc == DAOS_OC_OIT_RF3) ||
-	       (oc == DAOS_OC_OIT_RF4);
+	if (daos_obj_id2type(oid) == DAOS_OT_OIT)
```

Comment:
This should be

return daos_obi_id2type(oid) == DAOS_OT_OIT;

else 'return false' is missed